src/Common/src/TypeSystem/Common/TypeDesc.cs

```diff
@@ -99,7 +99,7 @@ public abstract partial class TypeDesc
         public override int GetHashCode()
         {
             // Inherited types are expected to override
-            return RuntimeHelpers.GetHashCode(this);
+            throw new NotImplementedException();
```

Comment:
Unfortunately, you can't do that as this needs to override the System.Object GetHashCode. (I REALLY don't want to have 2 different GetHashCode() functions to call which is what I could do.)